xbmc/interfaces/legacy/ListItem.cpp

```diff
@@ -398,7 +398,23 @@ namespace XBMCAddon
           else if (key == "set")
             item->GetVideoInfoTag()->m_strSet = value;
           else if (key == "imdbnumber")
-            item->GetVideoInfoTag()->m_strIMDBNumber = value;
+            item->GetVideoInfoTag()->SetUniqueID(value);
+          else if (key == "uniqueid")
+          {
+            if (alt.which() != second)
+              throw WrongTypeException("When using \"uniqueid\" you need to supply a list of tuples for the value in the dictionary");
+
+            std::map<std::string, std::string> uniqueids;
+            const std::vector<InfoLabelStringOrTuple>& listValue = alt.later();
+            for (std::vector<InfoLabelStringOrTuple>::const_iterator idEntry = listValue.begin(); idEntry != listValue.end(); ++idEntry)
+            {
+              if (idEntry->which() != second)
+                throw WrongTypeException("When using \"uniqueid\" you need to supply a list of tuples for the value in the dictionary");
+
+              uniqueids[idEntry->later().first()] = idEntry->later().second();
+            }
+            item->GetVideoInfoTag()->SetUniqueIDs(uniqueids);
+          }
```

Comment:
You need to add another `Alternative` to the signature. But given how long it already is, I say just drop this change and stick with the separate setter.